pkg/services/validationService.go

```diff
@@ -216,10 +216,44 @@ func (vs *ValidationService) Validate(admissionReviewReq *admission.AdmissionRev
 	return ParseEvaluationResponseIntoAdmissionReview(admissionReviewReq.Request.UID, allowed, msg, *warningMessages), false
 }
 
-type ClusterRequestMetadataAggregator = map[string]*cliClient.ClusterRequestMetadata
+func (m *clusterRequestMetadataMap) Len() int {
+	return m.entriesCount
+}
 
-var clusterRequestMetadataAggregator = make(ClusterRequestMetadataAggregator)
-var clusterRequestMetadataMutex = sync.RWMutex{}
+func (m *clusterRequestMetadataMap) Add(logJson string, clusterRequestMetadata *cliClient.ClusterRequestMetadata) {
+	m.clusterRequestMetadataAggregator.Store(logJson, clusterRequestMetadata)
+	m.entriesCount += 1
```

Comment:
if the item already exists, the `entriesCount` will be incremented, which is a mistake
```
if m.Load(logJson) == nil {
		m.entriesCount += 1
	}
```